src/material/button-toggle/button-toggle.ts

```diff
@@ -257,6 +269,45 @@ export class MatButtonToggleGroup implements ControlValueAccessor, OnInit, After
     this.disabled = isDisabled;
   }
 
+  /** Handle keydown event calling to single-select button toggle. */
+  protected _keydown(event: KeyboardEvent) {
+    if (this.multiple || this.disabled) {
+      return;
+    }
+
+    const target = event.target as HTMLButtonElement;
+    const buttonId = target.id;
+    const index = this._buttonToggles.toArray().findIndex(toggle => {
+      return toggle.buttonId === buttonId;
+    });
+
+    let nextButton;
+    switch (event.keyCode) {
+      case SPACE:
+      case ENTER:
+        nextButton = this._buttonToggles.get(index);
+        break;
+      case UP_ARROW:
+      case LEFT_ARROW:
+        nextButton = this._buttonToggles.get(
+          this._getNextIndex(index, this.dir === 'ltr' ? -1 : 1),
+        );
+        break;
+      case DOWN_ARROW:
+      case RIGHT_ARROW:
+        nextButton = this._buttonToggles.get(
+          this._getNextIndex(index, this.dir === 'ltr' ? 1 : -1),
```

Comment:
I think this will also invert the vertical arrows in RTL. We should only invert left/right.

Reply:
Understand. Updated in the new version.